### PR TITLE
Add embedded BC in RZ.

### PR DESCRIPTION
**Pull request by Yin-YinjianZhao**
In this PR, the embedded BC for electrostatic solver in RZ is added.
Previously, the embedded BC only worked for Cartesian.

### CODE REVIEW

File: Source/FieldSolver/ElectrostaticSolver.cpp [hunk 1]
```diff
@@ -342,14 +285,58 @@ WarpX::computePhiRZ (const amrex::Vector<std::unique_ptr<amrex::MultiFab> >& rho
         );
     }
 
+#ifndef AMREX_USE_EB
+
+    // Define the linear operator (Poisson operator)
+    MLNodeLaplacian linop( geom_scaled, boxArray(), DistributionMap() );
+
+#else
```
Comment (roelof-groenewald) [inline Source/FieldSolver/ElectrostaticSolver.cpp]: Is there a reason not to use the `MLEBNodeFDLaplacian` (used in the EB case) for the non-EB case as well? In recent tests I've done, the MLMG solver fails to converge when using `MLNodeLaplacian` but works fine with the EB version.

Comment (Yin-YinjianZhao) [reply Source/FieldSolver/ElectrostaticSolver.cpp]: @RemiLehe Maybe you have an answer for this question?

Comment (RemiLehe) [reply Source/FieldSolver/ElectrostaticSolver.cpp]: I think that this would be a question for @WeiqunZhang

**Comment by Yin-YinjianZhao** [comment]
An automated test will be added.
The results look like these two plots of phi and Er.
phi
Er
```
# This script tests the embedded boundary in RZ.
# A cylindrical surface (r=0.1) has a fixed potential 1 V.
# The outer surface has 0 V fixed.
# Thus the analytical solution has the form:
# phi(r) = A+B*log(r), Er(r) = -B/r.
```
([ipynb.txt](https://github.com/ECP-WarpX/WarpX/files/7736984/ipynb.txt) is used for plotting.)

File: Source/FieldSolver/ElectrostaticSolver.cpp (hunk 1 above)
Comment (RemiLehe) [reply Source/FieldSolver/ElectrostaticSolver.cpp]: Discussed with @WeiqunZhang today: `MLNodeLaplacian` should indeed be replaced by `MLEBNodeFDLaplacian` here.

Comment (RemiLehe) [reply Source/FieldSolver/ElectrostaticSolver.cpp]: @Yin-YinjianZhao Could you make the corresponding change?

File: Examples/Tests/ElectrostaticSphereEB/inputs_rz [hunk 2]
```diff
@@ -0,0 +1,30 @@
+max_step = 1
+amr.n_cell = 64 64
+amr.max_level = 0
+amr.blocking_factor = 8
+amr.max_grid_size = 128
+geometry.coord_sys   = 1
```
Comment (RemiLehe) [inline Examples/Tests/ElectrostaticSphereEB/inputs_rz]:
```suggestion
```

File: Examples/Tests/ElectrostaticSphereEB/analysis_rz.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+# Possible running time: < 1 s
+
+import os
+import sys
+
+import numpy as np
+import yt
+
+sys.path.insert(1, '../../../../warpx/Regression/Checksum/')
+#import checksumAPI
```
Comment (RemiLehe) [inline Examples/Tests/ElectrostaticSphereEB/analysis_rz.py]:
```suggestion
import checksumAPI
```

File: Examples/Tests/ElectrostaticSphereEB/analysis_rz.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+A = -B*np.log(0.5)
+
+err = 0.0
+errmax_phi = 0.0
+errmax_Er = 0.0
+for i in range(len(r)):
+    if r[i]>=0.1:
+        phi_theory = A+B*np.log(r[i])
+        Er_theory = -B/float(r[i])
+        err = abs( phi_theory - phi[i,phi.shape[1]//2] ) / phi_theory
```
Comment (RemiLehe) [inline Examples/Tests/ElectrostaticSphereEB/analysis_rz.py]: This only checks the fields in the center of the simulation. Maybe you could use this instead:
```suggestion
        err = abs( phi_theory - phi[i,:] ).max() / phi_theory
```

File: Examples/Tests/ElectrostaticSphereEB/analysis_rz.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        if err>errmax_phi:
+            errmax_phi = err
+        err = abs( Er_theory - Er[i,Er.shape[1]//2] ) / Er_theory
```
Comment (RemiLehe) [inline Examples/Tests/ElectrostaticSphereEB/analysis_rz.py]:
```suggestion
        err = abs( Er_theory - Er[i,:] ).max() / Er_theory
```

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**